### ISSUE TITLE
[lpcxpresso55s28] flash range starts from Secure address which is not compatible with latest Jlink(v7.82a)

### ISSUE THREAD

**Issue by hakehuang** [zephyrproject-rtos/zephyr#52190]
**Describe the bug**
in lpcxpresso55s28 dts config, starts from 0x10000000

build/zephyr/include/generated/autoconf.h:13:#define CONFIG_FLASH_BASE_ADDRESS 0x10000000
```
     iap: flash-controller@34000 {
                                ranges = < 0x0 0x10000000 0x3020000 >;

```
which is a Secure address and Jlink V7.82 does not support flash to this address. see below error

```
J-Link>loadfile "/home/shared/disk/zephyr_project/zephyr_lpcxpresso55s28/zephyr/samples/hello_world/build/zephyr/zephyr.bin" 0x10000000
'loadfile': Performing implicit reset & halt of MCU.
ResetTarget() start
Reset strategy selected in debugger ignored. Performing device specific reset.
Reset handler addr: 0x10001030
ResetTarget() end
AfterResetTarget() start
AfterResetTarget() end
Downloading file [/home/shared/disk/zephyr_project/zephyr_lpcxpresso55s28/zephyr/samples/hello_world/build/zephyr/zephyr.bin]...
Writing target memory failed.
```
and write to 0x0 works.

Please also mention any information which could help others to understand
the problem you're facing:
- all lpcxpresso55 series has the same problem

**To Reproduce**
1. west build -b lpcxpresso55s28
4.  west flash --runner=jlink

**Expected behavior**

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)
showstopper

**Logs and console output**
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)

**Environment (please complete the following information):**
 - OS: (e.g. Linux,)
 - Toolchain (e.g Zephyr SDK,)
 - Commit SHA or Version used: zephyr-v3.2.0-1553-g45e1ff94cd

**Comment by hakehuang** [comment]
@dleach02 , @mmahadevan108

**Comment by dleach02** [comment]
@hake, this appears to be a problem with all the different versions of jlink I've tried. If you change the device type to LPC5528JBD64 then it works. We haven't seen this before because the board is setup to use pyocd runner by default.

**Comment by dleach02** [comment]
Functionality of Segger changed with version v7.54b

**Comment by dleach02** [comment]
Segger confirmed the bug and will fix this in next week's release (11/21/2022)

**Comment by dleach02** [comment]
Confirmed fixed with v7.82d
